torch/csrc/distributed/rpc/future_message.h

```diff
@@ -12,20 +12,21 @@ namespace rpc {
 struct TORCH_API FutureMessage final {
  public:
   using Callback = std::function<void(const Message&)>;
+  FutureMessage() = default;
```

Comment:
Not a fan of default constructors. What's the motivation to add it?

Reply:
The default constructor was already there previously, but implicitly defined.
Adding the explicit single-argument constructor in this change seems to require explicitly declaring that an non-arg constructor (for an unrealized future) is allowed as well.